docs/tutorials/tutorial_1.rst

```diff
@@ -0,0 +1,448 @@
+
+7. All elements are positive/negative
+    This will reject solutions which do not properly handle empty collections for positive/negative elements.
+
+    ``CASE(N = 4, A = {1, 2, 3, 4});``
+
+    ``CASE(N = 4, A = {-1, -2, -3, -4});``
+
+8. All elements are zero
+    This can probably reject solutions which separate zeroes for some reasons (it's actually unnecessary).
+
+    This is already covered by (4).
+
+9. N = 2
+    This is the smallest possible value for N. Already covered by (2) and (6).
+
+10. Random values for elements of A
+     It's always good idea to include randomized input to the test data when the input space is very large (which should be true for most problems).
+
+     The randomized elements of A can be generated using a private function, as follows:
+
+     .. sourcecode:: cpp
+
+         void randomElements() {
+             for (int i = 0; i < N; i++) {
+                 A.push_back(rnd.nextInt(-1000000, 1000000));
+             }
+         }
+
+     (Note that at the beginning of the above method, N will have been set to a value from ``CASE()``, and A will have been cleared by ``BeforeTestCase()``.)
+
+     The above is good enough for this problem. However, it is nicer if we can somehow "tune" some properties of the randomization. For example, we can have parameters denoting the number of desired positive and negative numbers in A:
+
+
+     .. sourcecode:: cpp
+
+         void randomElements(int numPos, int numNeg) {
+             assert(numPos + numNeg <= N);
+
+             for (int i = 0; i < numPos; i++) {
+                 A.push_back(rnd.nextInt(1, 1000000));
+             }
+             for (int i = 0; i < numNeg; i++) {
+                 A.push_back(rnd.nextInt(-1000000, -1));
+             }
+             for (int i = 0; i < N-numPos-numNeg; i++) {
+                 A.push_back(0);
+             }
+
+             rnd.shuffle(A.begin(), A.end());
+         }
+
+     Again, the above tuning is not really necessary for this problem, as most tricky cases have been covered by previous hand-made test cases. However, for the purpose of learning, we will still use the tuning.
+
+     It is not the only tuning that we can do. Other options include:
+
+     - Tuning the range of possible values of the randomized element. For example, if we want the elements to be randomized between just 1 and 100.
+     - Same as above, but also tuning the percentage of the tuned range. For example, if we want the elements to be totally randomized, except a quarter of them, which should be between 1 and 100.
+     - etc.
+
+     All right, now use this function in ``CASE()``, with various arguments to it (and various values for N), for example as follows.
+
+     ``CASE(N = 10, randomElements(5, 5));``
+
+     ``CASE(N = 100, randomElements(20, 50));``
+
+     ``CASE(N = 1000, randomElements(300, 300));``
+
+     ``CASE(N = 10000, randomElements(2500, 6000));``
+
+
+11. N = 100000
```

Comment:
To make this point uniform with other points, probably need to explain the purpose of this point. (e.g. something like "To reject the solution which stores the integers in an array and the array size is less than 10^5")